testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -530,6 +532,37 @@ public void testUnionDistinctViews()
         }
     }
 
+    @Test(groups = {HIVE_ICEBERG_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testHiveIcebergView()
+    {
+        onTrino().executeQuery("DROP TABLE IF EXISTS iceberg.default.nation_iceberg_table");
+        onTrino().executeQuery("CREATE TABLE iceberg.default.nation_iceberg_table AS TABLE hive.default.nation");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS hive_iceberg_view");
+        onHive().executeQuery("CREATE VIEW hive_iceberg_view AS " +
+                "SELECT n_nationkey, n_name, r_regionkey, r_name " +
+                "FROM nation_iceberg_table JOIN region ON n_regionkey = r_regionkey");
+
+        // For an Iceberg table, the table schema in the metastore is generally ignored when reading directly from the table.
+        // In order to test that it's not used for Hive view translation, we desynchronize the state between metastore and the
+        // actual Iceberg schema in the storage. Using Hive connector here for convenience only.
+        onTrino().executeQuery("ALTER TABLE hive.default.nation_iceberg_table DROP COLUMN n_name");
```

Comment:
As discussed either do this drop via Hive (so we are sure the ALTER operation is not redirected to iceberg connector).
Or create another table with (BIGNIT dummy) schema and point it to share location of `hive_iceberg_view`. The schema of the other tables should still match the one of `hive_iceberg_view` as viewed by Trino, even if it is just `dummy` in HMS.